tests/io/test_io_image.py

```diff
@@ -142,3 +150,155 @@ def test_write_image(self, device, tmp_path, ext, channels):
+
+    @pytest.mark.parametrize("ext", ["tiff"])
+    def test_write_image_float32(self, device, tmp_path, ext):
+        """Test writing float32 images (TIFF only)."""
+        height, width = 4, 5
+        img_th: Tensor = create_random_img32_torch(height, width, 3, device)
+
+        file_path = tmp_path / f"image.{ext}"
+        write_image(file_path, img_th)
+
+        assert file_path.is_file()
+        # Note: Loading back float32 images may fail due to backend limitations
+        # We verify the file was written successfully, which is the main functionality
+
+    @pytest.mark.parametrize("quality", [None, 50, 80, 95])
+    def test_write_image_jpeg_quality(self, device, tmp_path, quality):
+        """Test that quality parameter works for JPEG files."""
+        height, width = 4, 5
+        img_th: Tensor = create_random_img8_torch(height, width, 3, device)
+
+        file_path = tmp_path / "image.jpg"
+        write_image(file_path, img_th, quality=quality)
+
+        assert file_path.is_file()
+        # Verify image can be loaded back
+        img_load = load_image(file_path, ImageLoadType.UNCHANGED)
+        assert img_load.shape == img_th.shape
+
+    @pytest.mark.parametrize("ext", ["png", "tiff"])
+    @pytest.mark.parametrize("quality", [None, 50, 95])
+    def test_write_image_non_jpeg_quality_ignored(self, device, tmp_path, ext, quality):
+        """Test that quality parameter is ignored for non-JPEG formats."""
+        height, width = 4, 5
+        img_th: Tensor = create_random_img8_torch(height, width, 3, device)
+
+        file_path = tmp_path / f"image.{ext}"
+        # Should work without errors even though quality is ignored
+        write_image(file_path, img_th, quality=quality)
+
+        assert file_path.is_file()
+        img_load = load_image(file_path, ImageLoadType.UNCHANGED)
+        assert img_load.shape == img_th.shape
+
+    def test_write_image_invalid_extension(self, device, tmp_path):
+        """Test that invalid file extensions raise an error."""
+        height, width = 4, 5
+        img_th: Tensor = create_random_img8_torch(height, width, 3, device)
+
+        file_path = tmp_path / "image.bmp"
+        with pytest.raises(Exception):  # KORNIA_CHECK raises an exception
+            write_image(file_path, img_th)
+
+    def test_write_image_invalid_shape(self, device, tmp_path):
+        """Test that invalid image shapes raise an error."""
+        # 1D tensor (invalid) - create using randint for uint8
+        img_th: Tensor = torch.randint(0, 256, (10,), device=device, dtype=torch.uint8)
+
+        file_path = tmp_path / "image.jpg"
+        with pytest.raises(Exception):  # KORNIA_CHECK raises an exception
+            write_image(file_path, img_th)
+
+    def test_write_image_invalid_dtype(self, device, tmp_path):
+        """Test that unsupported dtypes raise an error."""
+        height, width = 4, 5
+        # float64 is not supported
```

Comment:
Probably with custom errors types,  it how do you think we should support to report better errors to the user ?